### ISSUE TITLE
Allow clients to generate a new client secret.

### ISSUE THREAD

**Issue by khan-zia** [laravel/passport#1568]
Alright, so I looked this up in the repository before opening this issue and I think this hasn't been proposed and closed before so here I am proposing it.

I understand that most of the time clients are supposed to protect `secrets` and never lose them. However, I think there will be genuine use cases when a client wants to simply generate a fresh `secret` value without having to create an entirely new OAuth client. Examples include but are not limited to, parting ways with a company's employee that had access to the `secret` value, accidentally exposing the value, having a company-wide secrets rotation policy, etc.

It would be even better if client accounts can use more than one `secret` value to ensure zero downtime for their applications. This is until they replace the new `secret`, their app will be using the old one which they can safely discard later. However, this would be a big change and would require the database schema to change. Simply allowing a new `secret` to be generated would be a great start.

If you guys like the idea, I can put together a PR in the coming days. Cheers!

@taylorotwell @driesvints

**Comment by driesvints** [comment]
I'm not sure what the difference would be with generation a new OAuth client? The tokens related to the client would be invalid anyway so the old client can simply be closed off.

**Comment by khan-zia** [comment]
AFAIK, the tokens will not be invalidated if you simply update the `secret` value for a client. In fact, I just confirmed this before writing this comment. With this feature, I see the following benefits:

- It's just cumbersome to create a new client every time. You would be entering the OAuth values i.e. `redirect_uri` every time.
- This is especially cumbersome with heavily customized implementations. For e.g. we have overridden the `oauth_clients` model and require a dozen other inputs when creating a client.
- It saves the users of the client application from re-consent. If a new client is created, all its users will be forced to authorize it again.
- As I stated above, all existing tokens will continue to work.
